### PR TITLE
Prevent historical text rows from breaking board polling

### PR DESCRIPTION
**Pull request by cathrynlavery**
## Thinking Path

> - Paperclip is the control plane that keeps agent-company board polling, task lists, dashboard summaries, and heartbeat history available to operators.
> - The affected surface is the server-side list/read path for heartbeat runs, issues, and dashboard summary aggregates.
> - Historical local data can contain text values that are unsafe to extract, truncate, or aggregate directly in SQL when the database encoding/history is inconsistent.
> - A single bad historical row was enough to make board polling endpoints return 500, which made Paperclip appear broken even though most current records were valid.
> - This pull request moves risky text shaping out of SQL, bounds heartbeat payloads without SQL-side text extraction, and makes dashboard summary aggregation defensive.
> - The benefit is that one malformed historical text row no longer breaks the board polling endpoints that keep Paperclip usable.

## What Changed

- Kept heartbeat list responses bounded without SQL-side extraction of risky `result_json` text.
- Moved issue-list description truncation into TypeScript so SQL does not truncate untrusted historical text.
- Made dashboard summary aggregates defensive against malformed or empty rows.
- Added focused regression coverage for heartbeat list and dashboard summary behavior.
- Extended issue service regression coverage around list description handling.

## Verification

- `pnpm vitest run server/src/__tests__/dashboard-service.test.ts server/src/__tests__/heartbeat-list.test.ts`
- `pnpm vitest run server/src/__tests__/issues-service.test.ts`
- `pnpm --filter @paperclipai/server typecheck`
- `pnpm --filter @paperclipai/ui build`
- Manual local API check from the original reproducing branch: `/heartbeat-runs?limit=200` returned 200.
- Manual local API check from the original reproducing branch: `/issues?includeRoutineExecutions=true` returned 200.

## Risks

- Moderate scope risk because the change touches high-traffic board polling reads, but it is limited to response shaping and defensive aggregation.
- Heartbeat list payloads are intentionally bounded earlier, so consumers should not rely on list endpoints carrying full historical result payload text.
- No database migration or UI rendering change is included. Screenshots are not applicable.
- Full `pnpm test:run` was not run for this upstream branch; the focused regression tests, server typecheck, UI build, and original local endpoint checks above passed.

> For core feature work, check [`ROADMAP.md`](ROADMAP.md) first and discuss it in `#dev` before opening the PR. Feature PRs that overlap with planned core work may need to be redirected - check the roadmap first. See `CONTRIBUTING.md`.

## Model Used

- OpenAI Codex CLI, GPT-5.4, high reasoning, with local shell and GitHub CLI tool use.

## Checklist

- [x] I have included a thinking path that traces from project context to this change
- [x] I have specified the model used (with version and capability details)
- [x] I have checked ROADMAP.md and confirmed this PR does not duplicate planned core work
- [x] I have run tests locally and they pass
- [x] I have added or updated tests where applicable
- [x] If this change affects the UI, I have included before/after screenshots
- [x] I have updated relevant documentation to reflect my changes
- [x] I have considered and documented any risks above
- [x] I will address all Greptile and reviewer comments before requesting merge

### CODE REVIEW

File: server/src/services/heartbeat.ts [hunk 1]
```diff
@@ -412,15 +412,17 @@ const heartbeatRunListContextColumns = {
...
+  case
+    when ${heartbeatRuns.resultJson} is null then null
+    when pg_column_size(${heartbeatRuns.resultJson}) <= ${HEARTBEAT_RUN_SAFE_RESULT_JSON_MAX_BYTES}
+      then ${heartbeatRuns.resultJson}
+    else jsonb_build_object(
+      'truncated', true,
+      'truncationReason', 'oversized_result_json'
+    )
+  end
+`.as("resultJson");
```
[the remark this answers is not in the document]
Comment (cathrynlavery) [reply server/src/services/heartbeat.ts:827]: Added a code comment above the list resultJson projection documenting that small payloads intentionally return the complete JSONB object, while large payloads are stubbed to avoid SQL text extraction on historical client-encoding edge cases. Focused server tests still pass after the update.

File: server/src/services/issues.ts [hunk 2]
```diff
@@ -134,6 +134,14 @@ function sameRunLock(checkoutRunId: string | null, actorRunId: string | null) {
 const TERMINAL_HEARTBEAT_RUN_STATUSES = new Set(["succeeded", "failed", "cancelled", "timed_out"]);
 const ISSUE_LIST_DESCRIPTION_MAX_CHARS = 1200;
 
+function truncateIssueListDescription<T extends { description: string | null }>(row: T): T {
+  if (!row.description || row.description.length <= ISSUE_LIST_DESCRIPTION_MAX_CHARS) return row;
+  return {
+    ...row,
+    description: row.description.slice(0, ISSUE_LIST_DESCRIPTION_MAX_CHARS),
+  };
+}
```
[the remark this answers is not in the document]
Comment (cathrynlavery) [reply server/src/services/issues.ts:345]: Documented this tradeoff in code. I did not add an `octet_length`/SQL guard because the failure mode here is specifically SQL/client-encoding behavior around historical SQL_ASCII text rows; keeping truncation outside SQL is the safety boundary for this list endpoint. Focused server tests still pass after the update.

**Comment by cathrynlavery** [comment]
Hourly PR readiness sweep refreshed this branch against current `master` with a normal merge commit `a5fac4a`.

Verification:
- `git diff --check`
- focused existing tests already passing on the pre-refresh head: `pnpm --filter @paperclipai/server test server/src/__tests__/heartbeat-list.test.ts server/src/__tests__/dashboard-service.test.ts`

GitHub checks are running on the refreshed head now.

@greptile-apps please re-review the current head after the base refresh.

**Comment by cathrynlavery** [comment]
Addressed the CI failures exposed after the base refresh.

Fix commit: `cb87b409ea` (`fix: preserve issue descriptions after base refresh`)

What changed:
- Removed a stale base-refresh decode path that was treating plain issue descriptions as base64 and corrupting list responses.
- Kept the PR's intended app-side truncation path for issue list rows.

Verification:
- `git diff --check`
- `pnpm --filter @paperclipai/server test server/src/__tests__/heartbeat-list.test.ts server/src/__tests__/dashboard-service.test.ts server/src/__tests__/issue-blocker-attention.test.ts`
- `pnpm test:e2e tests/e2e/onboarding.spec.ts`

@greptile-apps please re-review the current head.
